diff a/bin/elasticsearch b/bin/elasticsearch
--- a/bin/elasticsearch
+++ b/bin/elasticsearch
@@ -72,7 +72,10 @@ while [ -h "$SCRIPT" ] ; do
 done

 # determine elasticsearch home
-ES_HOME=`dirname "$SCRIPT"`/..
+if [ -z "$ES_HOME" ]; then
+    echo "You must set the ES_HOME var" >&2
+    exit 1
+fi

 # make ELASTICSEARCH_HOME absolute
 ES_HOME=`cd "$ES_HOME"; pwd`
diff a/bin/elasticsearch.in.sh b/bin/elasticsearch.in.sh
--- a/bin/elasticsearch.in.sh
+++ b/bin/elasticsearch.in.sh
@@ -1,17 +1,5 @@
 #!/bin/sh

-# check in case a user was using this mechanism
-if [ "x$ES_CLASSPATH" != "x" ]; then
-    cat >&2 << EOF
-Error: Don't modify the classpath with ES_CLASSPATH. Best is to add
-additional elements via the plugin mechanism, or if code must really be
-added to the main classpath, add jars to lib/ (unsupported).
-EOF
-    exit 1
-fi
-
-ES_CLASSPATH="$ES_HOME/lib/elasticsearch-2.0.0-rc1.jar:$ES_HOME/lib/*"
-
 if [ "x$ES_MIN_MEM" = "x" ]; then
     ES_MIN_MEM=256m
 fi
diff a/bin/plugin b/bin/plugin
--- a/bin/plugin
+++ b/bin/plugin
@@ -16,7 +16,10 @@ while [ -h "$SCRIPT" ] ; do
 done

 # determine elasticsearch home
-ES_HOME=`dirname "$SCRIPT"`/..
+if [ -z "$ES_HOME" ]; then
+    echo "You must set the ES_HOME var" >&2
+    exit 1
+fi

 # make ELASTICSEARCH_HOME absolute
 ES_HOME=`cd "$ES_HOME"; pwd`
@@ -115,4 +118,4 @@ fi
 HOSTNAME=`hostname | cut -d. -f1`
 export HOSTNAME

-eval "$JAVA" -client -Delasticsearch -Des.path.home="\"$ES_HOME\"" $properties -cp "\"$ES_HOME/lib/*\"" org.elasticsearch.plugins.PluginManagerCliParser $args
+eval "$JAVA" -client -Delasticsearch -Des.path.home="\"$ES_HOME\"" $properties -cp "\"$ES_CLASSPATH/lib/*\"" org.elasticsearch.plugins.PluginManagerCliParser $args
